core/src/main/java/nl/nn/adapterframework/util/TransformerPool.java

```diff
@@ -154,6 +151,14 @@ public static TransformerPool getInstance(String xsltString, int xsltVersion) th
 		return new TransformerPool(xsltString, null, xsltVersion);
 	}
 
+	/**
+	 * Special utility method to create a new TransformerPool without a ClassLoader.
+	 * Utility pools should never use configuration classloaders, instead always read from the classpath!
+	 */
+	public static TransformerPool getUtilityInstance(String xsltString) throws TransformerConfigurationException {
+		return getUtilityInstance(xsltString, 0);
+	}
```

Comment:
De methode die je vervangt is gedeprecate omdat hij memory leaks (kan) veroorzaken bij verkeerd gebruik (zie #2555).
De utility instance moet geinstantieerd worden met NULL en niet `Thread.currentThread().getContextClassLoader()`.

UtilityInstances moeten eigenlijk weten wat de versie is, het is zonde die te moeten berekenen (middels een xslt), wanneer de xslt door ons wordt beheerd. Ik denk dat het verstandig is hier geen overloaded methode voor te maken en misschien dat we hier zelfs een check op kunnen doen? If version == 0 dan grote fout.

Ik zat net aan de SoapWrapper te denken, dat ding heeft wel even nodig om op te kunnen starten...

Reply:
De method die ik vervang accepteerde ook al een 0 parameter. Als dat eigenlijk niet zou moeten, moet ik deze method er dan weer uithalen en in de XmlValidator weer de deprecated method aanroepen?
Ik weet niet hoe ik vooraf de XSLT versie kan bepalen in de aanroepende code van de XmlValidator.

Reply:
Laten we het maar voor wat het is, als het goed is zou hij version 2.0 moeten pakken. We moeten sowieso een keer door die transformerpool klasse heen om hem op te schonen... Heel heel heel misschien dat je XmlUtils.DEFAULT_XSLT_VERSION kan gebruiken, maar dan moet je weer alles testen en dat is voor nu niet zo handig..

Reply:
Ik heb iig een issue gemaakt voor het opschonen van de getInstance methode #5501.